src/ui/public/resize_checker/resize_checker.js

```diff
@@ -0,0 +1,94 @@
+import $ from 'jquery';
+import ResizeObserver from 'resize-observer-polyfill';
+import { uniqueId, isEqual } from 'lodash';
+
+import EventsProvider from 'ui/events';
+
+export function ResizeCheckerProvider(Private) {
+  const EventEmitter = Private(EventsProvider);
+
+  function validateElArg(el) {
+    // the ResizeChecker historically accepted jquery elements,
+    // so we wrap in jQuery then extract the element
+    const $el = $(el);
+
+    if ($el.size() !== 1) {
+      throw new TypeError('ResizeChecker must be constructed with a single DOM element.');
+    }
+
+    return $el.get(0);
+  }
+
+  function getSize(el) {
+    return [el.clientWidth, el.clientHeight];
```

Comment:
Ah, I was thinking it ran on every resize event. Maybe it's not too bad then. It looks like `clientWidth` and `clientHeight` [do force layout](https://gist.github.com/paulirish/5d52fb081b3570c81e3a). Not sure how else to implement this resize ignoring method though. Maybe it would be worth adding a comment on `modifySizeWithoutTriggeringResize` saying "This method can trigger forced layouts, use with caution"?

I think this is a great PR btw! The forced layout thing just jumped out at me because I've been dealing with the same problem elsewhere in the code.